trajectory.py

```diff
@@ -248,15 +260,22 @@ def add_speed(self, overwrite=False):
         when using WGS 84 when it is calculated in metres. This is then divided
         by total seconds.
         """
-        if SPEED_COL_NAME in self.df.keys() and not overwrite:
+        if SPEED_COL_NAME in self.df.columns and not overwrite:
             raise RuntimeError('Trajectory already has speed values! Use overwrite=True to overwrite exiting values.')
-        self.df['prev_pt'] = self.df.geometry.shift()
-        self.df['t'] = self.df.index
-        self.df['delta_t'] = self.df['t'].diff()
+        self._update_prev_pt()
+        # NOTE: ? do we need the 't' again
+        #self.df = self.df.assign(t=self.df.index)
+        # self.df['delta_t'] = self.df['t'].diff()
+        if 't' in self.df.columns:
+            times = self.df.t
+        else:
+            times = self.df.reset_index().t
+        self.df = self.df.assign(delta_t=times.diff().values)
         try:
             self.df[SPEED_COL_NAME] = self.df.apply(self._compute_speed, axis=1)
         except ValueError as e:
             raise e
+        # ? what is that doing
```

Comment:
This sets the speed in the first row to the speed of the second row

Reply:
thanks for the explanation, maybe as a comment in source?